Purchases/Public/Purchases.swift

```diff
@@ -1366,30 +1362,52 @@ public extension Purchases {
                                              appUserID: String?,
                                              observerMode: Bool,
                                              userDefaults: UserDefaults?) -> Purchases {
-        configure(apiKey: apiKey,
-                  appUserID: appUserID,
-                  observerMode: observerMode,
-                  userDefaults: userDefaults,
-                  platformFlavor: nil,
-                  platformFlavorVersion: nil)
-    }
-
-    static internal func configure(apiKey: String,
-                                   appUserID: String?,
-                                   observerMode: Bool,
-                                   userDefaults: UserDefaults?,
-                                   platformFlavor: String?,
-                                   platformFlavorVersion: String?) -> Purchases {
+        configure(
+            withAPIKey: apiKey,
+            appUserID: appUserID,
+            observerMode: observerMode,
+            userDefaults: userDefaults,
+            useStoreKit2IfAvailable: false
+        )
+    }
+
+    /**
+     * Configures an instance of the Purchases SDK with a custom userDefaults. Use this constructor if you want to
+     * sync status across a shared container, such as between a host app and an extension. The instance of the
+     * Purchases SDK will be set as a singleton.
+     * You should access the singleton instance using ``Purchases.shared``
+     *
+     * - Parameter apiKey: The API Key generated for your app from https://app.revenuecat.com/
+     *
+     * - Parameter appUserID: The unique app user id for this user. This user id will allow users to share their
+     * purchases and subscriptions across devices. Pass `nil` or an empty string if you want `Purchases`
+     * to generate this for you.
+     *
+     * - Parameter observerMode: Set this to `true` if you have your own IAP implementation and want to use only
+     * RevenueCat's backend. Default is `false`.
+     *
+     * - Parameter userDefaults: Custom `UserDefaults` to use
+     *
+     * - Parameter useStoreKit2IfAvailable: opt in to using StoreKit 2 on devices that support it.
```

Comment:
maybe we can link from here for currently-supported storekit2 methods? at least while in beta?

Reply:
🤔 I guess it raises the question of "what exactly are we allowing for folks to choose between SK1 and SK2". 

For the time being I believe we're only going to provide the choice for purchases, since that's: 
- the most critical part
- the most battle-tested (albeit, somewhat buggy) API
- one where you actually _have_ a choice between SK1 and SK2 - for `beginRefundRequest`, `showManageSubscriptions` and `introEligibility`, SK1 and SK2 are not equivalent, with SK2 being superior for both `showManageSubscriptions` and `introEligibility`, and entirely required for `beginRefundRequest`. 

So, with that in mind, perhaps we should name the variable in a way that conveys that we're providing options for purchases only? 
i.e.: 
`useStoreKit2ForPurchasesIfAvailable`?

I'd love to know what others think though

Reply:
(that said, I'm approving the PR and happy to punt the discussion to a follow-up PR so as not to block this one)

Reply:
That makes sense, but will we consider adding more fine-grained configuration later on? If not, then I wonder if better docs for what this does in each version might be the way to go.

Reply:
I'm not sure. I can't think of any more fine-grained controls that would make sense off the top of my head, since once you go with purchases on a given SK version, that'll inform how you fetch products, observe the queue, etc. 

And I'm worried about making things too complicated for devs - at the end of the day, we'll know better which SK version to use for each scenario, I don't want to put the burden on devs to figure out what to do. 

I'd even say that we'd probably deprecate the flag once we've seen SK2 in production enough that we're confident in how it works.

Reply:
> I'd even say that we'd probably deprecate the flag once we've seen SK2 in production enough that we're confident in how it works.
👍

Reply:
fair point on what is actually "choosable" with this variable....though that begs the question -- do we want to check this variable from beginRefund/showManageSub/introElig? is there any reason a user would want the choice for those functions? i imagine the answer is no, just putting it out there to be sure we've considered it...

as for the docs--i'm just thinking that it's possible users don't have much knowledge of storekit2 at all when they come across this variable (a benefit of using revenuecat in the first place). so maybe even linking to docs about advantages/tradeoffs of using storekit2 (we had the blog post after the announcement...) would be a good idea? and then being clear if there isn't anything we DON'T currently support ,that a user could reasonably expect...

agreed, this doesn't have to happen on this PR. just something to consider before we release...

Reply:
> And I'm worried about making things too complicated for devs - at the end of the day, we'll know better which SK version to use for each scenario, I don't want to put the burden on devs to figure out what to do.

just read this after my previous comment, maybe I'm complicating things too much. 🤷‍♀️  just don't want developers to be like "what is THIS?!" and not have resources to learn about it.

Reply:
> as for the docs--i'm just thinking that it's possible users don't have much knowledge of storekit2 at all when they come across this variable

The parameter is optional (in the Objective-C sense hehe, meaning there's an overload of `configure` without it), so if they don't know what it is, they can ignore it.